src/core/entity/entity.hpp

```diff
@@ -62,14 +62,17 @@ class Entity : virtual public IEntity
         static const Entity::ProviderRulesDict providers{__VA_ARGS__};         \
         return providers;                                                      \
     }
-#define ENTITY_DECL_RELATION(dest, rule)                                       \
+
+#define ENTITY_DEF_RELATION(dest, rules)                                       \
+    Relation::build<dest>(getEntityManager().getEntity(getName()), rules)
+#define ENTITY_DEF_RELATION2(dest, rules)                                      \
```

Comment:
I guess the macro definitions required to be explained by documentation. There are two cases to acquire entity: by literal name and by specify with `template<>`; e.g.
``` getEntityManager().getEntity("Baseboard");```
``` getEntityManager().getEntity<Baseboard>();```

If we decide to explain the method of acquiring a target entity in the macro name, it will be too long a name. I decided to use postfix `2` to reduce macro name length. But you are right. It must be explained a little bit clearly.